api/src/main/java/jakarta/data/metamodel/constraint/Between.java

```diff
@@ -15,13 +15,9 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-package jakarta.data.metamodel.restrict;
+package jakarta.data.metamodel.constraint;
 
-public interface UnaryRestriction<T> extends Restriction<T> {
-    String attribute();
-
-    UnaryOperator operator();
-
-    @Override
-    UnaryRestriction<T> negate();
+public interface Between<T extends Comparable<T>> extends Constraint<T> {
```

Comment:
Why between is represented as with `GreaterThanOrEqual` and `LessThanOrEqual`? Some data sources might have a native between operation

Reply:
Well, sure, every SQL database has a native `between`, as does JDQL. But it's a very easy job for such databases to do this transformation.

Reply:
Having to do a transformation should only be a concern if an implementation were being asked to observe a pairing of GTE and LTE and identify this as an intended BETWEEN.  Prior to Gavin's changes we had that, but he actually got rid of that in this PR and switched it to `Between` as a first class `Constraint` type.  An implementation will identify BETWEEN by observing the `Between` subinterface the same way it would identify less than from a `LessThan` subinterface.

Reply:
@njr-11 I'm OK with between, it looks strange to me to include GTE and LTE in the between definition.

Reply:
> @njr-11 I'm OK with between, it looks strange to me to include GTE and LTE in the between definition.

Right, you have a good point there. This would be more straightforward as
```
    T lowerBound();
    T upperBound();
```
I was caught up on "Some data sources might have a native between operation" because I did not see how that mattered.

Reply:
I'm OK with that change. Originally the idea was that we would accommodating both open and closed intervals (an open interval being one with non-strict) bounds. But now that `Interval` was renamed to `Between`, and `GreaterThanOrEqual` was split from `GreaterThan`, this doesn't work out.

Reply:
Done.